pkg/storage/kubernetes.go

```diff
@@ -165,49 +206,11 @@ func (p *KubernetesIPPool) Allocations() []whereaboutstypes.IPReservation {
 
 // Update sets the pool allocated IP list to the given IP reservations
 func (p *KubernetesIPPool) Update(ctx context.Context, reservations []whereaboutstypes.IPReservation) error {
-	// marshal the current pool to serve as the base for the patch creation
-	orig := p.pool.DeepCopy()
-	origBytes, err := json.Marshal(orig)
-	if err != nil {
-		return err
-	}
-
-	// update the pool before marshalling once again
+	// update the pool with new ip allocations
 	p.pool.Spec.Allocations = toAllocationMap(reservations, p.firstIP)
-	modBytes, err := json.Marshal(p.pool)
-	if err != nil {
-		return err
-	}
-
-	// create the patch
-	patch, err := jsonpatch.CreatePatch(origBytes, modBytes)
-	if err != nil {
-		return err
-	}
-
-	// add additional tests to the patch
-	ops := []jsonpatch.Operation{
-		// ensure patch is applied to appropriate resource version only
-		{Operation: "test", Path: "/metadata/resourceVersion", Value: orig.ObjectMeta.ResourceVersion},
-	}
-	for _, o := range patch {
-		// safeguard add ops -- "add" will update existing paths, this "test" ensures the path is empty
-		if o.Operation == "add" {
-			var m map[string]interface{}
-			ops = append(ops, jsonpatch.Operation{Operation: "test", Path: o.Path, Value: m})
-		}
-	}
-	ops = append(ops, patch...)
-	patchData, err := json.Marshal(ops)
-	if err != nil {
-		return err
-	}
-
-	// apply the patch
-	err = p.client.Patch(ctx, orig, client.ConstantPatch(types.JSONPatchType, patchData))
+	err := p.client.Update(ctx, p.pool)
```

Comment:
With lock based approach, it is expected that only one client will update the ip pool at a time. So, json patch api should continue to be used instead of Update api.